chia/rpc/data_layer_rpc_api.py

```diff
@@ -298,3 +564,141 @@ async def get_kv_diff(self, request: Dict[str, Any]) -> EndpointResult:
         for rec in records:
             res.insert(0, {"type": rec.type.name, "key": rec.key.hex(), "value": rec.value.hex()})
         return {"diff": res}
+
+    # TODO: figure out the hinting
+    @marshal()  # type: ignore[arg-type]
+    async def make_offer(self, request: MakeOfferRequest) -> MakeOfferResponse:
+        # TODO: should the StoreProofs include the root?
+        maker_store_roots: Dict[bytes32, bytes32] = {}
+        maker_store_proofs: List[StoreProofs] = []
+        for offer_store in request.maker:
+            store_id = offer_store.store_id
+
+            # TODO: handle upserts?  deletes?
+            changelist = [
+                {
+                    "action": "insert",
+                    "key": entry.key,
+                    "value": entry.value,
+                }
+                for entry in offer_store.inclusions
+            ]
+            # TODO: am i reaching too far down here?  can't use DataLayer.batch_update()
+            #       since it publishes to chain.
+            new_root_hash = await self.service.data_store.insert_batch(
+                tree_id=offer_store.store_id,
+                changelist=changelist,
+            )
+            if new_root_hash is None:
+                raise Exception("only inserts are supported so a None root hash should not be possible")
+            maker_store_roots[store_id] = new_root_hash
+
+            proofs: List[Proof] = []
+            for entry in offer_store.inclusions:
+                # TODO: am i reaching too far down here?
+                node = await self.service.data_store.get_node_by_key(
+                    tree_id=store_id, key=entry.key, root_hash=new_root_hash
+                )
+                # TODO: gets nothing, maybe because the ancestors are not calculated
+                #       yet due to waiting for non-pending state?  maybe?
+                #       (see: 840910390980427)
+                proof_of_inclusion = await self.service.data_store.get_proof_of_inclusion_by_hash(
+                    node_hash=node.hash,
+                    tree_id=store_id,
+                )
```

Comment:
@fchirica, am I remembering correctly that this new root will be pending and will not have any ancestors?  If so, do you have any thoughts on handling that?  I'll try to give it some thought tomorrow.